core/src/test/java/org/elasticsearch/search/aggregations/bucket/children/ParentToChildrenAggregatorTests.java

```diff
@@ -0,0 +1,167 @@
+                createJoinField(PARENT_TYPE, id));
+    }
+
+    private static List<Field> createChildDocument(String childId, String parentId, int value) {
+        return Arrays.asList(new StringField(TypeFieldMapper.NAME, CHILD_TYPE, Field.Store.NO),
+                new StringField(UidFieldMapper.NAME, Uid.createUid(CHILD_TYPE, childId), Field.Store.NO),
+                new SortedNumericDocValuesField("number", value),
+                createJoinField(PARENT_TYPE, parentId));
+    }
+
+    private static SortedDocValuesField createJoinField(String parentType, String id) {
+        return new SortedDocValuesField(ParentFieldMapper.joinField(parentType), new BytesRef(id));
+    }
+
+    @Override
+    protected MapperService mapperServiceMock() {
+        MapperService mapperService = mock(MapperService.class);
+        DocumentMapper childDocMapper = mock(DocumentMapper.class);
+        DocumentMapper parentDocMapper = mock(DocumentMapper.class);
+        ParentFieldMapper parentFieldMapper = createParentFieldMapper();
+        when(childDocMapper.parentFieldMapper()).thenReturn(parentFieldMapper);
+        when(parentDocMapper.parentFieldMapper()).thenReturn(parentFieldMapper);
+        when(mapperService.documentMapper(CHILD_TYPE)).thenReturn(childDocMapper);
+        when(mapperService.documentMapper(PARENT_TYPE)).thenReturn(parentDocMapper);
+        when(mapperService.docMappers(false)).thenReturn(Arrays.asList(new DocumentMapper[] { childDocMapper, parentDocMapper }));
+        when(parentDocMapper.typeFilter()).thenReturn(new TypeFieldMapper.TypesQuery(new BytesRef(PARENT_TYPE)));
+        when(childDocMapper.typeFilter()).thenReturn(new TypeFieldMapper.TypesQuery(new BytesRef(CHILD_TYPE)));
+        return mapperService;
+    }
+
+    private static ParentFieldMapper createParentFieldMapper() {
+        Settings settings = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build();
+        return new ParentFieldMapper.Builder("parent").type(PARENT_TYPE).build(new Mapper.BuilderContext(settings, new ContentPath(0)));
+    }
+
+    private void testCase(Query query, CheckedConsumer<RandomIndexWriter, IOException> buildIndex, Consumer<InternalChildren> verify)
+            throws IOException {
+        Directory directory = newDirectory();
+        RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
+        buildIndex.accept(indexWriter);
+        indexWriter.close();
+
+        IndexReader indexReader = DirectoryReader.open(directory);
+        // TODO no "maybeWrap" for the searcher because this randomly led to java.lang.ClassCastException:
+        // org.apache.lucene.search.QueryUtils$FCInvisibleMultiReader cannot be
+        // cast to org.apache.lucene.index.DirectoryReader
+        // according to @mvg this can be fixed later but requires bigger changes
+        IndexSearcher indexSearcher = newSearcher(indexReader, false, true);
```

Comment:
@martijnvg since you know what needs to be done, can you open an issue for this? Then I can remove the comment and link to it so we don't forget to change this to include `maybeWrap` at some point